web/libs/editor/src/mixins/DynamicChildrenMixin.js

```diff
@@ -14,9 +15,15 @@ const DynamicChildrenMixin = types
     const prepareDynamicChildrenData = (data, store, parent) => {
       if (data && data.length) {
         for (const obj of data) {
+          // Multiple traversal of the tree will cause the same object to be added multiple times
+          if (parent.children?.find((child) => child.value === obj.value)) continue;
```

Comment:
This part makes me wonder, if we detect that the first child value exists, can we skip the remainder entirely? Or is it possible to have an incomplete tree?

Reply:
But after talking with Sergey we came up with even simpler check on the top of the method: `if (store.initialized) return;`. It will only be false within `initRoot()` call where we are propagating the values to tree nodes. All the other calls inside annotation's `afterCreate()` will use children, added before.
Also this check should only work for FF ON.

Reply:
Got it, will double check it tomorrow

Reply:
Had to make a few more changes but it should be ready to review again @hlomzik @Gondragos @bmartel.
Note: I added the FF check for extra safety but it is not needed

Reply:
Selenium passed, resolving